woocommerce/woocommerce-gutenberg-products-block
# Add chip component for coupons in the cart

This PR is part of #1786, but doesn't close it because during cooldown we need to roll out this component to the product filters. This PR only takes care of coupon styling on the cart page.

Introduces a "chip" component which are list items with remove icons. Takes some args to control the callback on remove, text, element, and the border radius.

<del>Note: Whilst working on this I realised some coupons couldn't be applied (coupon codes with special chars such as $). I fixed this in the API by moving the coupon code to the request body and updated tests. </del>

#### Accessibility




### Screenshots

![2020-02-25 15 05 29](https://user-images.githubusercontent.com/90977/75259823-89897800-57e0-11ea-8ad7-b06fe28dd6e9.gif)

### How to test the changes in this Pull Request:

1. Setup a cart block
2. Apply some coupons
3. Remove some coupons. Confirm display looks like above.
